Lesson30/havva/sort.js

```diff
@@ -0,0 +1,48 @@
+
+const exampleLetters = ['l', 'e', 'a', 'r', 'n'];
+const sortedArray = [...exampleLetters].sort();
+console.log('Sorted letters:', sortedArray);
+
+// Exercise 1:
+// Given an array of numbers, filter out the even numbers.
+const numbers = [3, 1, 4, 11, 5, 9, 2, 6, 5, 3, 5];
+
+console.log(numbers.sort((a, b) => a - b).filter(number => number % 2 === 0));
+
+// Exercise 2: Sorting Strings
+// Sort an array of strings in alphabetical order.
+// Expected output: ["apple", "banana", "grape", "orange", "pear"]
+const fruits = ["Banana", "apple", "Pear", "orange", "grape"];
+
+console.log(fruits, fruits.map(fruit => fruit.toLowerCase()).sort());
+
+
+// Exercise 3: Descending Order
+// Sort an array of numbers in descending order.
+// Expected output: [100, 40, 25, 10, 5, 1]
+const scores = [40, 100, 1, 5, 25, 10];
+
+console.log(scores.sort((a, b) => -(a - b)));
+console.log(scores.sort((a, b) => b - a)); // more efficient + direct way
+
+
+// Exercise 4: Sorting Objects
+// Sort an array of objects based on a specific property (e.g., age).
+// Expected output: Sorted array by age in ascending order
+const people = [
+  { name: "John", age: 30 },
+  { name: "Jane", age: 25 },
+  { name: "Gary", age: 35 },
+  { name: "Ellen", age: 22 }
+];
+
+console.log(people.map(person => person.age).sort((a, b) => a - b));
```

Comment:
This code extracts ages and sorts them, but doesn't sort the actual people objects as requested by the exercise. To sort the objects by age: people.sort((a, b) => a.age - b.age). sort() mutates the array, so use [...people].sort() if you need to preserve the original.



```suggestion
  console.log(people.sort((a, b) => a.age - b.age));
```